Library/Formula/llvm.rb

```diff
@@ -116,18 +152,39 @@ def install
     # Apple's libstdc++ is too old to build LLVM
     ENV.libcxx if ENV.compiler == :clang
 
-    if build.with?("lldb") && build.without?("clang")
-      raise "Building LLDB needs Clang support library."
+    # compiler-rt has some iOS simulator features that require i386 symbols
+    # I'm assuming the rest of clang needs support too for 32-bit compilation
+    # to work correctly, but if not, perhaps universal binaries could be
+    # limited to compiler-rt. llvm makes this somewhat easier because compiler-rt
+    # can almost be treated as an entirely different build from llvm.
+    ENV.permit_arch_flags
```

Comment:
Ran with/without ENV.permit_arch_flags. Without it, the build fails about here:
```
Scanning dependencies of target clang_rt.builtins_osx
/Applications/Xcode-beta.app/Contents/Developer/usr/bin/make -f projects/compiler-rt/lib/builtins/CMakeFiles/clang_rt.builtins_osx.dir/build.make projects/compiler-rt/lib/builtins/CMakeFiles/clang_rt.builtins_osx.dir/build
Generating libclang_rt.builtins_osx.a
cd /tmp/llvm20150923-33559-mwb0c7/projects/compiler-rt/lib/builtins && lipo -output /tmp/llvm20150923-33559-mwb0c7/./lib/clang/3.8.0/lib/darwin/libclang_rt.builtins_osx.a -create -arch i386 /tmp/llvm20150923-33559-mwb0c7/./lib/clang/3.8.0/lib/darwin/libclang_rt.builtins_i386_osx.a -arch x86_64 /tmp/llvm20150923-33559-mwb0c7/./lib/clang/3.8.0/lib/darwin/libclang_rt.builtins_x86_64_osx.a
fatal error: /Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/lipo: specifed architecture type (i386) for file (/tmp/llvm20150923-33559-mwb0c7/./lib/clang/3.8.0/lib/darwin/libclang_rt.builtins_i386_osx.a) does not match its cputype (16777223) and cpusubtype (3) (should be cputype (7) and cpusubtype (3))
make[2]: *** [projects/compiler-rt/lib/builtins/libclang_rt.builtins_osx.a] Error 1
make[1]: *** [projects/compiler-rt/lib/builtins/CMakeFiles/clang_rt.builtins_osx.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
With, it passes that point
```
Generating libclang_rt.builtins_osx.a
cd /tmp/llvm20150923-45191-1glhilv/projects/compiler-rt/lib/builtins && lipo -output /tmp/llvm20150923-45191-1glhilv/./lib/clang/3.8.0/lib/darwin/libclang_rt.builtins_osx.a -create -arch i386 /tmp/llvm20150923-45191-1glhilv/./lib/clang/3.8.0/lib/darwin/libclang_rt.builtins_i386_osx.a -arch x86_64 /tmp/llvm20150923-45191-1glhilv/./lib/clang/3.8.0/lib/darwin/libclang_rt.builtins_x86_64_osx.a
Built target clang_rt.builtins_osx
```
But fails later on
```
cd /tmp/llvm20150923-45191-1glhilv/projects/compiler-rt/lib/asan && /usr/local/Library/ENV/4.3/clang++   -DASAN_DYNAMIC=1 -DASAN_HAS_EXCEPTIONS=1 -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/tmp/llvm20150923-45191-1glhilv/projects/compiler-rt/lib/asan -I/tmp/llvm20150923-45191-1wmw25h/projects/compiler-rt/lib/asan -I/tmp/llvm20150923-45191-1glhilv/include -I/tmp/llvm20150923-45191-1wmw25h/include -I/tmp/llvm20150923-45191-1wmw25h/projects/compiler-rt/lib/asan/..  -fPIC -fvisibility-inlines-hidden -Wall -W -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wcovered-switch-default -Wnon-virtual-dtor -std=c++11 -Wall -std=c++11 -arch i386 -arch x86_64   -UNDEBUG   -fPIC -fvisibility-inlines-hidden -Wall -W -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wcovered-switch-default -Wnon-virtual-dtor -std=c++11 -Wall -std=c++11 -stdlib=libc++ -mios-simulator-version-min=7.0 -isysroot /Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator9.1.sdk -fPIC -fno-builtin -fno-exceptions -fomit-frame-pointer -funwind-tables -fno-stack-protector -fvisibility=hidden -fno-function-sections -fno-lto -O3 -gline-tables-only -Wno-gnu -Wno-variadic-macros -Wno-c99-extensions -Wno-non-virtual-dtor -fno-rtti -ftls-model=initial-exec -o CMakeFiles/RTAsan_dynamic.iossim.dir/asan_win.cc.o -c /tmp/llvm20150923-45191-1wmw25h/projects/compiler-rt/lib/asan/asan_win.cc
Undefined symbols for architecture i386:
  "_sigaltstack$UNIX2003", referenced from:
      __sanitizer::SetAlternateSignalStack() in sanitizer_posix_libcdep.cc.o
      __sanitizer::UnsetAlternateSignalStack() in sanitizer_posix_libcdep.cc.o
  "_sleep$UNIX2003", referenced from:
      __sanitizer::SleepForSeconds(int) in sanitizer_posix_libcdep.cc.o
  "_usleep$UNIX2003", referenced from:
      __sanitizer::SleepForMillis(int) in sanitizer_posix_libcdep.cc.o
  "_waitpid$UNIX2003", referenced from:
      __sanitizer::SymbolizerProcess::StartSymbolizerSubprocess() in sanitizer_symbolizer_posix_libcdep.cc.o
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [lib/clang/3.8.0/lib/darwin/libclang_rt.ubsan_iossim_dynamic.dylib] Error 1
```

Manual builds outside of homebrew complete, though, so maybe ENV.permit_arch_flags is addressing a symptom rather than a cause...